### DISCUSSION TITLE
Having issues with reset({}, {keepValues: true}); in 7.13+

### DISCUSSION THREAD

**Discussion by LagMuffle** [react-hook-form/react-hook-form#6483] [Q&A]
I am utilising FormProvider to create a single-route Component based faux multi section form. On submission of the current section it hides the section and displays the next section of the form. The user is able to navigate back and forward across sections with the form still populated and the values populated on the input fields. On final submission the entire form contents is outputted to an email.

Each section of the form features yup validation and a revalidation mode of onChange; to enable the revalidation of each section, the submit button submits the entire form, and to clear the revalidation mode from bleeding across when switching across form sections I was using calling reset({},{keepValues:true}); in UseEffect on every submission to maintain the form content but remove the revalidation mode flag so it no longer appeared on the next section, if it was called in the previous one.

Since updating the react-hook-form package to 7.13 (and above) this no longer works, with that reset call now simply resetting the previous data and clearing all the fields. Removing the reset function allows for the data to persist across sections but I can no longer remove the onChange validation. It works as expected if I roll back t 7.12.

I dug through the changes to the package on 7.13 and couldnt find a cause. Is anyone aware of why this might be happening now or is there a better way to remove the reValidateMode after a form submission?

**Comment by bluebill1049** [comment]
can you share a simple codesandbox?

**Reply by LagMuffle** [reply]
[Codesandbox](https://codesandbox.io/s/dry-cookies-nn3lf) This is functioning as exepected in 7.12.2, updating react-hook-forms anymore causes it to fall over.

**Reply by bluebill1049** [reply]
1. provide a small codesandbox so i can easily reproduce the problem
2. provide steps to reproduce the issue.

**Reply by LagMuffle** [reply]
Apologies.

[Small Sandbox](https://codesandbox.io/s/react-hook-form-reset-keep-values-796wk)

**Working example:** 
- Enter input into box and click next
- Form resets ReValidation at each section
- Values are maintained when moving back/forward between sections
- Information for all sections is displayed as an alert when submitting section 3

**Issue:** 
- Values not being maintained with **reset({}, {keepValues: true})** when updating react-hook-form above 7.12

**Steps to reproduce:** 
- Update sandbox react-hook-form package above 7.12.2  
- Step through form as per working example

**Issue Outcome:** 
- Values are lost after moving between sections of the form

Expected Outcome: 
- Values will persist after **reset({}, {keepValues: true})** is called.

**Reply by bluebill1049** [reply]
thanks for the detailed update. haveing a look at it now.

**Reply by bluebill1049** [reply]
Track it here: https://github.com/react-hook-form/react-hook-form/issues/6513

**Comment by bluebill1049** [comment]
I think i got your problem now, because your inputs are getting unmounted, so it's important to not reset default values as well: 

let me know if this is what you expected: 
https://codesandbox.io/s/clever-swirles-f1s23?file=/src/App.js

**Reply by LagMuffle** [reply]
Unfortunately I cannot get your codesandbox to work as desired in either the most recent version, or by rolling back to 7.12.2 as I was able to get a working solution before, with the hidden input's value now always being removed on reset if it is not on screen and then re added.

In my example I was leveraging which doesn't appear to be unmounting the inputs when they are toggled via a state change in 7.12, allowing for the values to remain on reset. Is this something that was changed in 7.13?

**Reply by LagMuffle** [reply]
*leveraging FormProvider

**Reply by bluebill1049** [reply]
> with the hidden input's value now always being removed on reset if it is not on screen and then re added.

can you update my codesandbox above?

**Reply by LagMuffle** [reply]
Switching the import in your [codesandbox] (https://codesandbox.io/s/summer-grass-6n1jc) above from your up-to-date "src" dir and calling the "react-hook-form" dependency @7.12 produces the desired effect, with the value in the second input remaining intact while the forms logic is reset while the input is either hidden and unhidden.

**Reply by bluebill1049** [reply]
looking into it.

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/issues/6513 re-open and getting it fixed.
